src/components/SpiceList.js

```diff
@@ -7,12 +7,34 @@ class SpiceList extends React.Component {
     search: ""
   }
 
+
+  // Ian -- can you help me with this below? It seems too involved....
   renderSpices() {
-    return this.props.spices.map(spice => (
+    let filteredSpices = this.props.spices.filter(spice => {
+      if (this.state.search === "" && this.state.fourStarOnly === false) {
+        return true
+      } else if (this.state.search !== "" && this.state.fourStarOnly === false) {
+       return spice.notes.toLowerCase().includes(this.state.search.toLowerCase())
+      } else if (this.state.search === "" && this.state.fourStarOnly === true) {
+        return spice.rating >= 4
+      } else {
+        return spice.rating >= 4 && spice.notes.toLowerCase().includes(this.state.search.toLowerCase())
+      }
+    })
```

Comment:
When you've got multiple filter conditions for an array, it can help to use separate filter methods to organize your code instead of doing it all in one. Here's an approach of chaining multiple filters together:

```js
let filteredSpices = this.props.spices
      .filter(spice => this.state.fourStarOnly ? spice.rating >= 4 : true)
      .filter(spice => spice.notes.toLowerCase().includes(this.state.search.toLowerCase()))
```